tag/nitag.yml

```diff
@@ -0,0 +1,1814 @@
+        200:
+          description: Success
+        400:
+          description: Bad Request
+        401:
+          $ref: '#/responses/Unauthorized'
+        default:
+          $ref: '#/responses/Error'
+  /v2/tags/{path}/update-values:
+    post:
+      tags:
+        - tags
+      summary: Send multiple value updates to a tag
+      description: Push a list of current value update objects to update the value of the tag at the path multiple times. This can be used to send value history in bulk.
+      operationId: update-tag-values
+      responses:
+        200:
+          description: Tag Values Updated
+        400:
+          description: Bad Request
+        401:
+          $ref: '#/responses/Unauthorized'
+        409:
+          description: Conflict
+        default:
+          $ref: '#/responses/Error'
+      parameters:
+        - in: path
+          name: path
+          description: Tag path
+          type: string
+          required: true
+        - in: body
+          name: tagValues
+          description: Timestamped values
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/TagCurrentValue'
+          required: true
+
+definitions:
+  Error:
+    description: Contains error information
+    type: object
+    properties:
+      name:
+        description: String error code
+        type: string
+      code:
+        description: Numeric error code
+        type: integer
+      resourceType:
+        description: Type of resource associated with the error
+        type: string
+      resourceId:
+        description: Identifier of the resource associated with the error
+        type: string
+      message:
+        description: Complete error message
+        type: string
+      args:
+        description: Positional argument values for the error code
+        type: array
+        items:
+          type: string
+      innerErrors:
+        type: array
+        items:
+          $ref: '#/definitions/Error'
+    example:
+      name: Tag.NoSuchTag
+      code: -251909
+      resourceType:
+      resourceId:
+      message: "No such tag: 'foo'"
+      args: ['foo']
+  TagType:
+    type: string
+    title: Tag Data Type
+    enum: &TYPES [DOUBLE, INT, STRING, BOOLEAN, U_INT64, DATE_TIME]
+  TagWithValue:
+    description: A tag and its current value
+    type: object
+    title: Tag With Value
+    properties:
+      tag:
+        $ref: '#/definitions/Tag'
+      current:
+        $ref: '#/definitions/TagCurrentValue'
+      aggregates:
+        $ref: '#/definitions/TagAggregates'
+    example:
+      tag:
+        path: foo
+        type: DOUBLE
+        keywords: [fooKeyword, barKeyword]
+      current:
+        value:
+          type: DOUBLE
+          value: "3.14"
+        timestamp: '2018-09-04T18:45:08Z'
+      aggregates:
+        sum: "10.5"
+        min: "0.0"
+        max: "5.0"
+        mean: "2.0"
+        count: 5
+  TagValue:
+    description: Value of a tag
```

Comment:
What's the difference between this and current value of a tag below?

Is this one just used in operations that deal with past values?

Reply:
This one just contains the value and the type, while TagCurrentValue also contains a timestamp. This one is just used in contexts where a route only returns a value and a type, as some of the routes provide support for projecting out timestamps. Thanks for pointing this out though, as I don't like the name "TagCurrentValue". I changed it to "TimestampedTagValue", as that makes it more clear.